internal/service/streamconnection/data_source_stream_connections_test.go

```diff
@@ -0,0 +1,92 @@
+package streamconnection_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"go.mongodb.org/atlas-sdk/v20231115002/admin"
+)
+
+func TestAccStreamDSStreamConnections_basic(t *testing.T) {
+	var (
+		orgID          = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName    = acctest.RandomWithPrefix("test-acc-stream")
+		instanceName   = acctest.RandomWithPrefix("test-acc-instance")
+		dataSourceName = "data.mongodbatlas_stream_connections.test"
+	)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBetaFlag(t); acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             CheckDestroyStreamConnection,
+		Steps: []resource.TestStep{
+			{
+				Config: streamConnectionsDataSourceConfig(kafkaStreamConnectionConfig(orgID, projectName, instanceName, "user", "rawpassword", "localhost:9092,localhost:9092", "earliest", false)),
```

Comment:
[q - not blocking] Since all the data source tests use a resource. Couldn't we just have one acceptance test file where we have data source and resource test together? I see how this would decrease the amount of resources that we need to run the acceptance tests.

Reply:
It is a good point, I did go for this option in the case of search_deployment resource as execution times were quite long.
For this case the creation/update/delete of instance connections is very fast, so saw no harm in leaving in separate tests. One test case, Cluster connection type, does require creating a cluster but when running locally you can reference existing an existing cluster using env variables.